### PR TITLE
Fix s3MetaToAzureProperties Content-Md5 key

### PR DESCRIPTION
**Pull request by jmaitrehenry**
## Description
The key was not correctly checked because we `CanonicalHeaderKey`ise it just before

## Motivation and Context
Fix a part of #4994 

## How Has This Been Tested?
Testing by putting some new object on Azure

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/gateway-azure.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func s3MetaToAzureProperties(s3Metadata map[string]string) (storage.BlobMetadata
 		case k == "Content-Length":
 			// assume this doesn't fail
 			props.ContentLength, _ = strconv.ParseInt(v, 10, 64)
-		case k == "Content-MD5":
+		case k == "Content-Md5":
```
Comment (harshavardhana) [inline cmd/gateway-azure.go:103]: Can you add a unit test for this?

**Comment by jmaitrehenry** [comment]
I will try :)
Minio is my first project in Go and I want to learn more. It could take a little time before having a test working. (Or maybe not, it will be my first test in go)

**Comment by jmaitrehenry** [comment]
@harshavardhana It's really easy to add a test finally

File: cmd/gateway-azure_test.go [hunk 2]
```diff
@@ -82,6 +82,14 @@ func TestS3MetaToAzureProperties(t *testing.T) {
 			t.Fatalf("Test failed with unexpected error %s, expected UnsupportedMetadata", err)
 		}
 	}
+
+	headers = map[string]string{
+		"Content-MD5": "d41d8cd98f00b204e9800998ecf8427e",
+	}
+	_, props, _ := s3MetaToAzureProperties(headers)
+	if props.ContentMD5 != headers["Content-MD5"] {
```
Comment (harshavardhana) [inline cmd/gateway-azure_test.go]: This should have failed right?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by jmaitrehenry** [comment]
Without the fix, it failed.
The http header is Content-MD5 and should works.

**Comment by harshavardhana** [comment]
> Without the fix, it failed.
> The http header is Content-MD5 and should works.

Yes but the fix is looking for `Content-Md5` not `Content-MD5`  looks like its being converted using http.CanonicalHeaderKey - perhaps test should have `Content-Md5` not `Content-MD5` to avoid confusion. Also `Content-Md5` should be base64 not `hex` of md5sum.

**Comment by jmaitrehenry** [comment]
The fix look for the `CanonicalHeaderKey` version of the header.
The tests check for the right header sent by client `Content-MD5 ` wich is transformed to `Content-Md5` by minio, and set to `props.ContentMD5` for Azure blob.

> Also Content-Md5 should be base64 not hex of md5sum
Right, I could change it for a real md5 for Azure, but, most client will send a md5 in "8209ed86ac433cee4e5218e0d93b7d6c" format.

I could add something that recognize this kind of md5 and trasnform it for azure.

**Comment by harshavardhana** [comment]
Content-MD5 is never sent by the client if it's following HTTP RFC. Go http server also marshals the errant key names like this into proper canonical names in req.Header.

So our transformation test rather looks unexpected that is why we can keep it simply `content-md5` since all metadata map entries are small cases.

**Comment by jmaitrehenry** [comment]
@harshavardhana fixed

**Review by harshavardhana — APPROVED** [review]

File: cmd/gateway-azure_test.go [hunk 3]
```diff
@@ -82,6 +82,14 @@ func TestS3MetaToAzureProperties(t *testing.T) {
 			t.Fatalf("Test failed with unexpected error %s, expected UnsupportedMetadata", err)
 		}
 	}
+
+	headers = map[string]string{
+		"content-md5": "Dce7bmCX61zvxzP5QmfelQ==",
+	}
+	_, props, _ := s3MetaToAzureProperties(headers)
```
Comment (balamurugana) [inline cmd/gateway-azure_test.go]: You would need to get and check error also

**Review by balamurugana — CHANGES_REQUESTED** [review]
minor error in test case

**Review by balamurugana — APPROVED** [review]
As discussed https://github.com/minio/minio/pull/5068/files/caf8aff52e189e62543f9738b9815af2f4a709da#diff-2ee5453b2351c76bb4c5b94d087d5a96R89 can be taken into separate PR

File: cmd/gateway-azure_test.go (hunk 3 above)
Comment (jmaitrehenry) [reply cmd/gateway-azure_test.go]: Will do tonight

**Comment by jmaitrehenry** [comment]
Sorry for the delay @balamurugana, it's now fixed.

**Review by balamurugana — APPROVED** [review]

**Comment by deekoder** [comment]
Ignoring codecoverage as spurious. Test has been added. Merging hence.

**Pull request merged**